### ISSUE TITLE
Unnecessary identity change triggered

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#6792]
```
2019-01-21T10:47:27.591923327Z level=info msg="Identity of endpoint changed" containerID=f5678a1465 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=83 identity=34866 identityLabels="k8s:app=prometheus,k8s:component=core,k8s:io.cilium.k8s.policy.cluster=default,k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s,k8s:io.kubernetes.pod.namespace=monitoring" ipv4=10.10.1.155 ipv6="f00d::a0a:100:0:f26b" k8sPodName=monitoring/prometheus-core-5c9fbf5cc6-h874w oldIdentity=34866 subsys=endpoint
```

**Comment by ianvernon** [comment]
@tgraf can you provide more context here?

**Comment by jrajahalme** [comment]
So the new identity is the same as the old one. We have two callers of `endpoint.SetIdentity()` out of which the restoring one seems like the more likely culprit. I'll add a check to avoid the `SetIdentity()` call if there is no change.

**Comment by jrajahalme** [comment]
This fix to this is so close to the fix for #6793 that these ought to be tested together. So both fixed in #6937.

**Comment by tgraf** [comment]
Fixed by #6937
